Assets/Scripts/Noise/PerlinNoise.cs

```diff
@@ -0,0 +1,210 @@
+/// <summary>
+/// Using Perlin Noise function from adrian's soapbox "Understanding Perlin Noise" by Flafla2.
+/// https://flafla2.github.io/2014/08/09/perlinnoise.html
+/// From August 9th, 2014
+/// 
+/// github link: https://gist.github.com/Flafla2/f0260a861be0ebdeef76
+/// </summary>
+public class PerlinNoise {
+    /// <summary>
+    /// Doubled permutation to avoid overflow.
+    /// </summary>
+    private int[] p;
```

Comment:
p is not a good variable name, perhaps permutationOverflow or permutationLimit. Even private variables should be name properly to aid readability.

Reply:
Renamed to permutation :)